test/cri-containerd/container_fileshare_test.go

```diff
@@ -0,0 +1,198 @@
+//go:build functional
+
+package cri_containerd
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	"github.com/Microsoft/hcsshim/pkg/annotations"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+func Test_Container_File_Share_Writable_WCOW(t *testing.T) {
+	requireFeatures(t, featureWCOWHypervisor)
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+
+	client := newTestRuntimeClient(t)
+	ctx := context.Background()
+
+	sbRequest := getRunPodSandboxRequest(t, wcowHypervisorRuntimeHandler,
+		WithSandboxAnnotations(map[string]string{
+			annotations.DisableWriteableFileShares: "true",
+		}),
+	)
+	podID := runPodSandbox(t, client, ctx, sbRequest)
+	defer removePodSandbox(t, client, ctx, podID)
+	defer stopPodSandbox(t, client, ctx, podID)
+
+	var (
+		tempDir       = t.TempDir()
+		containerPath = "C:\\test"
+		testFile      = "t.txt"
+		testContent   = "hello world"
+	)
+	ioutil.WriteFile(
+		filepath.Join(tempDir, testFile),
+		[]byte(testContent),
+		0644,
+	)
+
+	cRequest := &runtime.CreateContainerRequest{
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name(),
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageWindowsNanoserver,
+			},
+			Command: []string{
+				"cmd",
+				"/c",
+				"ping -t 127.0.0.1",
+			},
+			Mounts: []*runtime.Mount{
+				{
+					HostPath:      tempDir,
+					ContainerPath: containerPath,
+					Readonly:      false,
+				},
+			},
+		},
+		PodSandboxId:  podID,
+		SandboxConfig: sbRequest.Config,
+	}
+	cID := createContainer(t, client, ctx, cRequest)
+	defer removeContainer(t, client, ctx, cID)
+
+	// container should fail because of writable mount
+	_, err := client.StartContainer(ctx, &runtime.StartContainerRequest{ContainerId: cID})
+	// error is serialized over gRPC then embedded into "rpc error: code = %s desc = %s"
+	//  so error.Is() wont work
+	if !strings.Contains(err.Error(), fmt.Errorf("adding writable shares is denied: %w", hcs.ErrOperationDenied).Error()) {
+		if err == nil {
+			stopContainer(t, client, ctx, cID)
+		}
```

Comment:
can we move this out of this outer error check?

Reply:
as in:
```go
if err == nil {
    stopContainer(t, client, ctx, cID)
}
if !strings.Contains(err.Error(), ... ) {
    t.Fatalf()
}
```

Reply:
yep!